### PR TITLE
HHH-13000 - Fix for refreshing entities under PESSIMISTIC_WRITE lock

### PR DESCRIPTION
**Pull request by vladmihalcea**
https://hibernate.atlassian.net/browse/HHH-13000

### CODE REVIEW

**Comment by vladmihalcea** [comment]
@JWGmeligMeyling @beikov I created this PR to replace [the original one](https://github.com/hibernate/hibernate-orm/pull/2551). The same logic should apply to `PESSIMISTIC_READ` which, on DBs that don't really support this mode, Hibernate falls back to `PESSIMISTIC_WRITE`.

If you agree with this commit, I can integrate it after your review.

File: hibernate-core/src/main/java/org/hibernate/event/internal/DefaultRefreshEventListener.java [hunk 1]
```diff
@@ -194,7 +196,7 @@ public void onRefresh(RefreshEvent event, Map refreshedAlready) {
 					lockOptionsToUse.setLockMode( LockMode.READ );
 
 					// and prepare to reset the entry lock-mode to WRITE after the refresh completes
```
Comment (gsmet) [inline hibernate-core/src/main/java/org/hibernate/event/internal/DefaultRefreshEventListener.java]: Let's update the comment. The lock mode won't always be WRITE.

**Comment by gsmet** [comment]
@vladmihalcea It makes sense. Please update the comment before merging, thanks!

**Comment by gsmet** [comment]
Fixed the comment and merged. Thanks all!

**Comment by JWGmeligMeyling** [comment]
Thanks for merging! Is this a candidate for backport too? Or won't there be a 5.3 release anymore?

**Comment by gsmet** [comment]
No plan to backport it to 5.3. It’s now in a product so the bar to get into 5.3 is very high. 

That’s why we’re working on releasing a 5.4 soon so that we can get all these fixes released.

**Pull request closed without merging**